### ISSUE TITLE
[Discussion] Storyline!

### ISSUE THREAD

**Issue by WardBenjamin** [TeamPorcupine/ProjectPorcupine#431]
For a while we've had some kind of basic story outline (See #263), something along the lines of ![Story](https://cloud.githubusercontent.com/assets/4739008/17813852/11a32762-65fb-11e6-856a-c555c9988f41.PNG)

Basically the concept is that you get screwed over by QuillCorp, and stranded with no fuel and only a basic drill and power source, and your smallish station.

Any discussion? What does everyone think about this?

Let's get the ball rollin'!

**Comment by bjubes** [comment]
I thought we were using the link in the readme to disuse future topic like this.

**Comment by bjubes** [comment]
http://feathub.com/TeamPorcupine/ProjectPorcupine/

**Comment by vogonistic** [comment]
I love it. What kind of mechanics would we need to make a story mode?

We should let modders add storydriven gameplay where you have goals and an event chain, but also randomized events that has some depth. We should implement any campaign/story mode as an internal mod so that we know the API is available.

@WardBenjamin Can you add this to feathub?

**Comment by WardBenjamin** [comment]
Yep! Added!

**Comment by Tranberry** [comment]
@bjubes isn't that more for suggesting ideas not to discuss them?

**Comment by vogonistic** [comment]
@Tranberry The feathub votes will establish a roadmap, it's pretty crucial to get things on there if we want them prioritized

**Comment by Tranberry** [comment]
@vogonistic  Some topics there are lacking good descriptions, I'm probably a bit nit picky but an idea is barley an idea when it just says "What about aliens". **But if the idea is to 'brainstorm' ideas that later will be fleshed out I kind of understand it.** I'm fine with it either way I just didn't get it.

**Comment by vogonistic** [comment]
I'm not sure which topic you are talking about, but ideally the better fleshed out one's will win. The problem right now is I think people will want everything. I wish we had something where people could vote for 3 items, and 3 only.

**Comment by Tranberry** [comment]
@vogonistic I'm going way off topic trying to reply here. I guess I'm just in agreement with you.

But this will be useful for making a game tutorial right, I mean a scenario is not that far off from a tutorial anyhow.

**Comment by lincore81** [comment]
In my opinion df-likes don't benefit from a story. Yes, there is a premise: three pawns stranded on a rim world/strike the earth. The premise is only a frame for the true story, which is told in terms of game mechanics, an emergent narrative. Who was crushed by elephants, whose loved one got shot, how did the colony of pacifists survive etc.

Of course, there are also games like Theme Hospital, which consist of a campaign of multiple scenarios. Here, a story could be useful to connect the scenarios in one narrative. However, I personally prefer Dwarf Fortress's way of giving you an empty canvas to draw inappropriate things on. 

I think I prefer Rimworld way of giving control over the premise to the players via scenarios. If all starting conditions were defined via xml/lua, one could easily create alternative scenarios - with alternative premises.

**Comment by b4ux1t3** [comment]
@lincore81 I think your last paragraph about sums up what I think. The base game could have a simple backstory (Space miners stranded! More at 11!), but having specific scenarios that you can play through (and that you can create for _others_ to play through) would be hugely beneficial.

**Comment by ajahraus** [comment]
It seems to me that a 'story' (i.e. specific turn of events that lead the characters to the place they are now) is much less important than a 'setting' (i.e. the universe and its history that the current events unfold within). 
Are the space-pirates members of a naturally violent race, or were they a peaceful and resource rich race before some great injustice left them scattered, weak, and hungry for a return to power?
Are robotic aliens an ancient and incomprehensible threat (i.e. Rimworld's mechanoids), or are they autonomous mining drones built by a moral-less corporation with little regard for collateral damage? I personally like the idea of a low level threat being a mining drone attacking your ship simply because it can't tell the difference between a space rock and a space ship. 
This of course leads to the question, are alien races pre-made and hand crafted to fit a particular theme, or are they procedural? I think hand crafting them allows for more control of themes and emergent story (i.e. dwarves, elves, humans, goblins in DF), but a procedural system could lead to some really cool replay-ability later on.

**Comment by Tranberry** [comment]
I'm the opposite of you, I want a game to be as story driven as Crusader Kings, Europa Universalis, Dwarf Fortress and many other -in these games what you do and what happens makes the story. 

This means that the game have to have a more refined system of interlocking game mechanics to not make the game seem like the same as last time you played it. That said, I absolutely think a starter scenario would be great for people to learn the game and so and why not have a static story for that?

**Comment by mikejbrown** [comment]
@mattomwit Welcome! @Tranberry With a proper scenario system we could accommodate both of you! Even better if the scenarios are moddable. That probably won't be for v0.1 though...

**Comment by Tranberry** [comment]
yeah I'm dreaming of v2.1. This is just my ideal outcome of the project.

**Comment by kd7uiy** [comment]
See also #1078 #1076  and #1070 for some of the game play discussions that have happened on this subject.

**Comment by Raketfart** [comment]
I think this intro text adds a lot of flavor. It doesn't have to dictate the goal of the game as revenge, but you could have missions or trade opportunities, that were references to greedy quillcorp.

**Comment by Tranberry** [comment]
While it's a **very important** topic, is this really vital for v0.1? 

https://github.com/TeamPorcupine/ProjectPorcupine/wiki/Milestone-0.1

**Comment by kd7uiy** [comment]
I say let's close this one, but add the other ones that I referenced to the list of topics we need to review for 0.1. They are what I was really thinking of for the 0.1 milestone. I'm okay with leaving this issue opened too, BTW.

**Comment by TomMalbran** [comment]
We should have a Wiki page for things for future milestones so that we can add them to a milestone when we want to, but still have them documented somewhere.
